### ISSUE TITLE
Some questions about the center loss

### ISSUE THREAD

**Issue by taylover-pei** [KaiyangZhou/pytorch-center-loss#7]
Congratuations! You have done a great work. But I have a question about the center loss.
Can you explain why you wrote there two lines?
https://github.com/KaiyangZhou/pytorch-center-loss/blob/082ffa21c065426843f26129be51bb1cfd554806/main.py#L128
Thank you very much! Looking forward to your reply!

**Comment by devraj89** [comment]
What I understood is that the author has said the weight parameter will be used to compute the total loss. But when the updation of the centers is being done it is being done without the weight parameter.
Basically, the updates for the centers in this code is nothing but the backpropagation of the mean square error loss.
